2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,398 @@
+
+    private static final Predicate<Field> HEAD_FIELD = field -> field
+            .getType().getSimpleName().equals(NODE_NAME)
+            & (field.getName().contains("head")
+            | field.getName().contains("first"));
+
+    private static final Predicate<Field> TAIL_FIELD = field -> field
+            .getType().getSimpleName().equals(NODE_NAME)
+            & (field.getName().contains("tail")
+            | field.getName().contains("last"));
 
     private Queue<Integer> integerQueue = new LinkedQueue<>();
 
+
     @Test
     @Order(1)
-    void sizeOfEmptyQueue() {
-        assertEquals(0, integerQueue.size());
+    void createNodeClass() {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        String name = innerClass.getSimpleName();
+
+        assertThat(name).isEqualTo(NODE_NAME);
     }
 
     @Test
     @Order(2)
-    void size() {
-        integerQueue.add(98);
-        integerQueue.add(9);
-        integerQueue.add(5);
-        integerQueue.add(6);
+    void chekFieldsNameInNodeClass() {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        boolean hasElementField = hasField(innerClass, ELEMENT_FIELD);
+        boolean hasNodeField = hasField(innerClass, NODE_FIELD);
 
-        assertEquals(4, integerQueue.size());
+        assertThat(hasElementField).isTrue();
+        assertThat(hasNodeField).isTrue();
     }
 
     @Test
     @Order(3)
-    void addElementIntoEmptyQueue() {
-        integerQueue.add(1312);
+    void checkFieldsInQueueCLass() {
+        Class<?> baseClass = this.integerQueue.getClass();
+        boolean hasHeadField = hasField(baseClass, HEAD_FIELD);
+        boolean hasSizeFiled = hasField(baseClass, SIZE_FIELD);
+        boolean hasTailFiled = hasField(baseClass, TAIL_FIELD);
 
-        assertEquals(1, integerQueue.size());
-        assertEquals(1312, integerQueue.poll().intValue());
+        assertThat(hasHeadField).isTrue();
+        assertThat(hasSizeFiled).isTrue();
+        assertThat(hasTailFiled).isTrue();
     }
 
     @Test
     @Order(4)
-    void addElement() {
-        integerQueue.add(324);
-        integerQueue.add(23);
-        integerQueue.add(5);
+    void sizeReturnsZeroWhenQueueIsEmpty() {
+        int size = getInternalSize();
 
-        assertEquals(3, integerQueue.size());
-        assertEquals(324, integerQueue.poll().intValue());
+        assertThat(size).isEqualTo(0);
     }
 
     @Test
     @Order(5)
-    void pollElementFromEmptyQueue() {
-        assertNull(integerQueue.poll());
+    void addRisesSize() {
+        addIntElementToQueue(1);
+        addIntElementToQueue(2);
+        addIntElementToQueue(3);
+        int size = getInternalSize();
+
+        assertThat(size).isEqualTo(3);
     }
 
     @Test
-    @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
+    @Order(5)
+    void addExpensesSize() {
```

Comment:
Suppose this method must have name `pollExpensesSisze()` and use `integerQueue.pop()` method instead of reflection.

Reply:
This concerns to all the poll tests.